src/solver/simulation/sim_alloc_probleme_hebdo.cpp

```diff
@@ -46,307 +46,270 @@ void SIM_AllocationProblemeHebdo(PROBLEME_HEBDO& problem, uint NombreDePasDeTemp
-    problem.IndexDebutIntercoExtremite = (int*)MemAlloc(nbPays * sizeof(int));
-    problem.IndexSuivantIntercoOrigine = (int*)MemAlloc(linkCount * sizeof(int));
-    problem.IndexSuivantIntercoExtremite = (int*)MemAlloc(linkCount * sizeof(int));
-    problem.NumeroDeJourDuPasDeTemps = (int*)MemAlloc(NombreDePasDeTemps * sizeof(int));
-    problem.NumeroDIntervalleOptimiseDuPasDeTemps
-      = (int*)MemAlloc(NombreDePasDeTemps * sizeof(int));
-    problem.NbGrpCourbeGuide = (int*)MemAlloc(NombreDePasDeTemps * sizeof(int));
-    problem.NbGrpOpt = (int*)MemAlloc(NombreDePasDeTemps * sizeof(int));
-    problem.CoutDeDefaillancePositive = (double*)MemAlloc(nbPays * sizeof(double));
-    problem.CoutDeDefaillanceNegative = (double*)MemAlloc(nbPays * sizeof(double));
-    problem.CoutDeDefaillanceEnReserve = (double*)MemAlloc(nbPays * sizeof(double));
-    problem.NumeroDeContrainteEnergieHydraulique = (int*)MemAlloc(nbPays * sizeof(int));
-    problem.NumeroDeContrainteMinEnergieHydraulique = (int*)MemAlloc(nbPays * sizeof(int));
-    problem.NumeroDeContrainteMaxEnergieHydraulique = (int*)MemAlloc(nbPays * sizeof(int));
-    problem.NumeroDeContrainteMaxPompage = (int*)MemAlloc(nbPays * sizeof(int));
-    problem.NumeroDeContrainteDeSoldeDEchange = (int*)MemAlloc(nbPays * sizeof(int));
-
-    problem.NumeroDeContrainteEquivalenceStockFinal = (int*)MemAlloc(nbPays * sizeof(int));
-    problem.NumeroDeContrainteExpressionStockFinal = (int*)MemAlloc(nbPays * sizeof(int));
-
-    problem.NumeroDeVariableStockFinal = (int*)MemAlloc(nbPays * sizeof(int));
-    problem.NumeroDeVariableDeTrancheDeStock = (int**)MemAlloc(nbPays * sizeof(int*));
-    for (uint p = 0; p < nbPays; ++p)
-        problem.NumeroDeVariableDeTrancheDeStock[p] = (int*)MemAlloc(100 * sizeof(int));
-
-    problem.ValeursDeNTC
-      = (VALEURS_DE_NTC_ET_RESISTANCES**)MemAlloc(NombreDePasDeTemps * sizeof(void*));
-    problem.ValeursDeNTCRef
-      = (VALEURS_DE_NTC_ET_RESISTANCES**)MemAlloc(NombreDePasDeTemps * sizeof(void*));
-    problem.ConsommationsAbattues
-      = (CONSOMMATIONS_ABATTUES**)MemAlloc(NombreDePasDeTemps * sizeof(void*));
-    problem.ConsommationsAbattuesRef
-      = (CONSOMMATIONS_ABATTUES**)MemAlloc(NombreDePasDeTemps * sizeof(void*));
-    problem.AllMustRunGeneration
-      = (ALL_MUST_RUN_GENERATION**)MemAlloc(NombreDePasDeTemps * sizeof(void*));
-    problem.SoldeMoyenHoraire
-      = (SOLDE_MOYEN_DES_ECHANGES**)MemAlloc(NombreDePasDeTemps * sizeof(void*));
+    problem.DefaillanceNegativeUtiliserPMinThermique.assign(nbPays, false);
+    problem.DefaillanceNegativeUtiliserHydro.assign(nbPays, false);
+    problem.DefaillanceNegativeUtiliserConsoAbattue.assign(nbPays, false);
+
+    problem.CoefficientEcretementPMaxHydraulique.assign(nbPays, 0.);
+
+    problem.BruitSurCoutHydraulique.assign(nbPays, std::vector<double>(8784));
+
+    problem.NomsDesPays = std::vector<const char*>(nbPays);
```

Comment:
About : `problem.NomsDesPays = std::vector<const char*>(nbPays);`
Why not : 
```C++
problem.NomsDesPays.resize(nbPays);
```
? 
Same remark for other assignment like this